packages/frontend/core/src/components/affine/setting-modal/general-setting/editor/edgeless/snapshot.tsx

```diff
@@ -54,14 +93,35 @@ export const EdgelessSnapshot = (props: Props) => {
       });
     });
 
+    // append to dom node
     wrapperRef.current.append(editorHost);
-  }, [docName]);
+  }, [docName, editorSetting, keyName, updateElement]);
 
   useEffect(() => {
     // eslint-disable-next-line @typescript-eslint/no-floating-promises
     renderEditor();
   }, [renderEditor]);
 
+  // observe editor settings change
+  useEffect(() => {
+    editorSetting.provider
+      .watchAll()
+      .pipe(
+        map(settings => {
+          if (typeof settings[keyName] === 'string') {
+            return JSON.parse(settings[keyName]);
+          }
+          return keyName;
+        }),
+        pairwise()
+      )
+      .subscribe(([prev, current]) => {
```

Comment:
it seems this subscription never get unsubscribed

Reply:
Thanks! I forgot to unsubscribe the subscription. Code updated.